src/v/cloud_storage/read_path_probes.cc

```diff
@@ -19,7 +19,7 @@
 namespace cloud_storage {
 
 partition_probe::partition_probe(const model::ntp& ntp) {
-    if (config::shard_local_cfg().disable_metrics()) {
+    if (!config::shard_local_cfg().disable_metrics()) {
         return;
     }
```

Comment:
this reads oddly to me. wouldn't we return from the probe construction if metrics are disabled?

Reply:
that was an accident
great catch